### ISSUE TITLE
AnnotationSpec does not find tests declared in constructor

### ISSUE THREAD

**Issue by rollie42** [kotest/kotest#1215]
Simply, 

```
class MyTests : AnnotationSpec() {
  @Test
  fun someTest() {...}
}
```

works as expected, but 

```
class MyTests : AnnotationSpec({
  @Test
  fun someTest() {...}
})
```

finds no tests and elicits no warnings. I think this format is supposed to work as well? Or if not, it's a really easy mistake to make when changing from another spec to `AnnotationSpec`.

**Comment by sksamuel** [comment]
AnnotationSpec is meant as a "gateway drug" to the framework from junit. So having the constructor-body style doesn't make much sense, and probably is impossible to find local tests anyway. I'll remove this ability from AnnotationSpec.

**Comment by sksamuel** [comment]
Fixed in master (4.0 snapshot).
